check/controls.go

```
@@ -79,12 +74,88 @@ func NewControls(in []byte, definitions []string) (*Controls, error) {
 	return c, nil
 }
 
+func (controls *Controls) convertAuditToRegisteredType( auditType AuditType, audit interface{} ) (Auditer, error) {
+
+	if auditType == "" || auditType == TypeAudit {
+
+		if s, ok := audit.(string); ok || audit == nil {
+			return &Audit{audit: s}, nil
+		}
+		return nil, fmt.Errorf("failed to convert audit, mismatching type", ) // TODO add Check ID here
+	} else {
+
+		if callback, ok := controls.auditTypeRegistry[auditType]; ok {
+			o := callback()
+			if auditAsString, e := yaml.Marshal(audit); e == nil {
+
+				if e := yaml.Unmarshal(auditAsString, o); e != nil {
+					return nil, fmt.Errorf("unable to Unmarshal Audit %v", e)
+
+				}
+				return o.(Auditer), nil // we don't check error because it already verified in convertAuditToRegisteredType
+
+			}else { // 'else' clause used here due to 'e' variable scope
+				return nil, fmt.Errorf("unable to marshal Audit %v", e)
+			}
+		}
+		return nil, fmt.Errorf("audit type %v is not registered", auditType)
+	}
+}
+
+func (controls *Controls) RegisterAuditType(auditType AuditType, typeCallback func()interface{}) error  {
+
+	if _,ok :=controls.auditTypeRegistry[auditType]; !ok{
```

Handle the error case (in this case it's the "ok" branch) first so you can return the error code and then get back to a left-aligned happy path.

---

Also missing space after := here